scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -583,6 +583,7 @@ class ScioContext private[scio] (
 
   private[scio] def execute(): ScioExecutionContext = {
     val sc = this
+    testId.foreach(tId => TestDataManager.overrideTransforms(tId, pipeline))
```

Comment:
```suggestion
    if (isTest) {
      TestDataManager.overrideTransforms(tId, pipeline)
    }
```
I know it's functionally the same but this makes the logic more clear imo.

Reply:
We need to unwrap the actual testId, so it'd need to be a pattern match on the option with an empty `None` case ... which should just be a `foreach`. Or use `testId.get` which 🙅

Reply:
heh, you'll hate this but there is a lot of precedent for `testId.get` in Scio 😅  I just think it makes the intention of what's going on clearer.

Reply:
😢 Done.